tests/fixtures/pipeline_objects.py

```diff
@@ -191,10 +313,24 @@ def make_datastore(*args, cache_dir=None, cache_base_name=None, session=None):
             if ds.image is None:  # make the preprocessed image
                 _logger.debug('making preprocessed image. ')
                 ds = preprocessor.run(ds)
+                if bad_pixel_map is not None:
+                    ds.image.flags |= bad_pixel_map
+                    if ds.image.weight is not None:
+                        ds.image.weight[bad_pixel_map.astype(bool)] = 0.0
```

Comment:
You want to use ds.image.flags here rather than bad_pixel_map, as the former is the one that's bitwise ORed both of them together.